### PR TITLE
Move static data from config to resources/config/

### PR DESCRIPTION
**Pull request by LukasReschke**
This moves the static data from config into resources/config/ to prevent further update problems.

@rullzer Please review with regard to the mimetype stuff
@karlitschek FYI

### CODE REVIEW

**Comment by Xenopathic** [comment]
We should keep the mimetypealiases.json and mimetypemapping.json in config/, since those files are supposed to be edited by the admin. The .dist.json files can go in resources/ though

**Comment by karlitschek** [comment]
I disagree. People will forget to add the mimetypemapping.json the same way the miss the cert. We have to move this over too.
Changing the file is a very esoteric usecase. Why should an admin do this?
Actually I suggest to go into the oposite direction. ownCloud shouldn't try to detect and handle all the mimetypes in the first place. This will lead to performance problems and has no benefit for the user. The only places where we need to know the mimetypes is to show the right icon in the UI and to filter for gallery and muic. But this is only a subset of the mimetypes that don't change at all.

**Comment by karlitschek** [comment]
Can we move it to core/ressources? I think we shold have less crap in the root directory. Maybe sometime we can cleanup more stuff like moving the db_structure.xml and other to /core too.
Beside that :+1:

**Comment by Xenopathic** [comment]
@karlitschek Note, the file doesn't exist by default. It's for admins that wish to add custom mimetype aliases or mappings. ownCloud distributes a standard set of mimetype aliases/mappings,  which currently go in `config/mimetypealiases.dist.json` and `config/mimetypemapping.dist.json`, which I agree should go in `/resources`. But the admin-configurable files should stay in `config/`

cc @rullzer

**Comment by rullzer** [comment]
@karlitschek basically what @Xenopathic said. I think the dist one should go into `/resources` but the user one should stay in `/config`. The user generated one is not the be shipped by default. So only if an admin has a very specific use case it adds it. And then it should go into `/config`. That way it will persist during upgrades etc.

Just to be clear. The file in `/config` basically extends the one in `/resources`. So if an admin does not add it ownCloud will just use the shipped one in `/resources`

**Comment by karlitschek** [comment]
Got it. But why should an admin extend this list in the first place? What is the usecase?

**Comment by rullzer** [comment]
@karlitschek well if the default mapping is not good enough. Lets say I want all my .cpp files to have a custom icon because I'm a huge C++ fanboy.

Or if a hospital has some obscure format to store large medical images in. Then they might want to map the correct extensions to that mimetype. And at the same time map that mimetype to a generic image icon.

**Comment by karlitschek** [comment]
So how would you actually add the mimetype icons? In a theme?
Again. This is a very esoteric usecase.

**Comment by rullzer** [comment]
Yes if you have specific icons you would add them in a theme.
But even without new icons there might file extensions that you want mapped to certain mimetypes which are just not the case by default. (For example the .dvi extension is not mapped by default).

Most of the admins will not create those custom mappings or aliases. And that is totally fine. But there are legit use cases where we should not limit them and make them patch core. A config file is a lot cleaner.

But I think this really is besides the point. If there is user configured stuff it should go into `/config`. The stuff bundled with owncloud by default (our default mappings and mimetype aliases) they should go into `/resources`.

**Comment by LukasReschke** [comment]
Fair enough… Then let's for whatever reason somebody might want to modify mimetypes keep these manual edits in config :see_no_evil:

**Comment by LukasReschke** [comment]
@Xenopathic @rullzer Happier now? - If so please review :speak_no_evil:

File: lib/private/files/type/detection.php [hunk 1]
```diff
@@ -102,10 +111,10 @@ private function loadAliases() {
 			return;
 		}
 
-		$this->mimeTypeAlias = json_decode(file_get_contents($this->configDir . '/mimetypealiases.dist.json'), true);
+		$this->mimeTypeAlias = json_decode(file_get_contents($this->defaultConfigDir . '/mimetypealiases.dist.json'), true);
```
Comment (Xenopathic) [inline lib/private/files/type/detection.php]: We can probably drop the `.dist` part if it's not in `config/`. That was only there to differentiate between the distributed file and the one that the admin can change.

**Comment by Xenopathic** [comment]
Code looks good :+1:

Now it's a race between this and #19272 for the merge conflict :laughing:

**Comment by LukasReschke** [comment]
THIS THIS THIS THIS THIS! ;-)

**Comment by Xenopathic** [comment]
@LukasReschke But you *like* merge conflicts, right? I wouldn't want to take that away from you :laughing: Think of it as an early Christmas present

**Pull request merged**